### PR TITLE
Add a new extension to search confluence wiki

### PR DESCRIPTION
**Pull request by daviddkkim**
## Description

This is a new extension that allows users to search Confluence Wiki.
They would authenticate by specifying their domain, username and api token. Afterwards, they can search pages as if they are on the browser. 

You can either navigate to the page or copy URL to clipboard.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/12561526/149828901-04b39c07-53b2-499e-88a7-cff25fdf8f5f.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/confluence-search/package.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  "name": "confluence-search",
+  "title": "Confluence Search",
+  "description": "Search through Confluence",
+  "icon": "confluence-icon.png",
+  "author": "daviddkkim",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Confluence Wiki",
```
Comment (pernielsentikaer) [inline extensions/confluence-search/package.json:12]:
```suggestion
      "title": "Search Wiki",
      "subtitle": "Confluence",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @daviddkkim 👋 

Thanks for your contribution 💯 

I don't have credentials to test it fully, do you by any chance have some credentials I could borrow a few minutes?

One thing I noticed is that when wrong credentials are present, it doesn't die gracefully (showing a toast and stopping the loader with an empty array) – Can you look into this?

v6zs1oQBhRdViP1J

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by daviddkkim** [comment]
Screen capture including the authentication

https://user-images.githubusercontent.com/12561526/150695809-0132ed52-5380-4910-81c3-1631c51a46bc.mp4

**Comment by daviddkkim** [comment]
@pernielsentikaer 

Made changes based on our slack convo:
- removed unecessary await 
- added in proper loading state based on whether the API call has returned rather than list item length
- changed pref field name from domain url to domain.


The empty list is there to show empty list in case API call response status was not OK. (fail state).

**Comment by pernielsentikaer** [comment]
I sent you some more feedback in our conversation in Slack, let me know if you need anything 😄

**Comment by daviddkkim** [comment]
@pernielsentikaer changed made !

File: extensions/confluence-search/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      }
+    />
+  );
+}
+
+enum Status {
+  Failure,
+  Success,
+}
+
```
Comment (pernielsentikaer) [inline extensions/confluence-search/src/index.tsx:116]:
```suggestion
```
Unused enum status

File: extensions/confluence-search/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            <OpenInBrowserAction title="Open in Browser" url={confluenceUrl + "/wiki" + searchResult.url} />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
+            <CopyToClipboardAction
+              title="Copy URL"
+              content={confluenceUrl + "/wiki" + searchResult.url}
+              shortcut={{ modifiers: ["cmd"], key: "." }}
+            />
+          </ActionPanel.Section>
+        </ActionPanel>
```
Comment (pernielsentikaer) [inline extensions/confluence-search/src/index.tsx:106]:
```suggestion
        <ActionPanel>
          <ActionPanel.Section>
            <OpenInBrowserAction title="Open in Browser" url={confluenceUrl + "/wiki" + searchResult.url} />
            <CopyToClipboardAction
              title="Copy URL"
              content={confluenceUrl + "/wiki" + searchResult.url}
              shortcut={{ modifiers: ["cmd"], key: "." }}
            />
          </ActionPanel.Section>
        </ActionPanel>
```
You could put it in the same section

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @daviddkkim 👋 

Awesome, thanks for the corrections 💯  Only two small NITs back and we're ready to proceed 💪

**Comment by daviddkkim** [comment]
> Hi @daviddkkim 👋
> 
> Awesome, thanks for the corrections 💯 Only two small NITs back and we're ready to proceed 💪

Thanks Per!

**Review by pernielsentikaer — APPROVED** [review]
Awesome @daviddkkim 💪 

Thanks for your contribution 💯 I'm going to accept it and then will my colleague do a final review before it's finally merged.

**Comment by sxn** [comment]
🎉

**Pull request merged**